### PR TITLE
Sage300: PostOnboarding Fixes - 2

### PR DESCRIPTION
**Pull request by anishfyle**
## Summary by CodeRabbit

- **New Features**
  - Enhanced search functionality to return more accurate reference numbers based on specific types.
  - Improved user interface for mapping configurations, with conditional rendering for better usability.
  - New event handling for export log dialog visibility, improving user interaction.

- **Bug Fixes**
  - Fixed an issue where errors were not displayed correctly in the dashboard error section.
  - Corrected the conditional logic for displaying export summaries to reflect updated data properties.

- **Documentation**
  - Updated user-facing texts and messages to provide clearer information on export progress and statuses.

- **Style**
  - Added new styles for the search bar and dialog components to improve visual interaction.

- **Refactor**
  - Renamed properties in the accounting export summary model for better clarity.
  - Streamlined the logic in several components to enhance performance and readability.

- **Chores**
  - Updated import statements to reflect changes in the enum model.

**Note:** Internal code-level details have been omitted to maintain confidentiality.

### CODE REVIEW

**Comment by anishfyle** [comment]
@coderabbitai review

File: src/app/shared/components/helper/mapping/generic-mapping-v2/generic-mapping-v2.component.html [hunk 1]
```diff
@@ -1,7 +1,7 @@
 <div *ngIf="isLoading" class="tw-flex tw-justify-center tw-items-center tw-h-screen">
     <app-loader></app-loader>
 </div>
-<div *ngIf="!isLoading">
+<div *ngIf="!isLoading && !isMappingLoading">
         <div>
             <app-card-mapping-header (triggerAutoMapEmployee)="triggerAutoMapEmployees()" [sourceField]="sourceField" [showAutoMapEmployee]="showAutoMapEmployee" [mappingStats]="mappingStats" (triggerAutoMapEmployee)="triggerAutoMapEmployees"></app-card-mapping-header>
```
[the remark this answers is not in the document]
Comment (ashwin1111) [reply src/app/shared/components/helper/mapping/generic-mapping-v2/generic-mapping-v2.component.html:6]: +1 @anishfyle

File: src/app/shared/components/dashboard/dashboard-export-summary-section/dashboard-export-summary-section.component.scss [hunk 2]
```diff
@@ -6,4 +6,8 @@
     align-items: flex-start;
     gap: 16px;
     max-width: auto;
+}
+
+.pointer {
+    cursor: pointer;
```
Comment (ashwin1111) [inline src/app/shared/components/dashboard/dashboard-export-summary-section/dashboard-export-summary-section.component.scss:12]: tailwind, whole file

File: src/app/shared/components/helper/mapping/generic-mapping-v2/generic-mapping-v2.component.ts [hunk 3]
```diff
@@ -29,6 +29,8 @@ export class GenericMappingV2Component implements OnInit {
 
   @Input() destinationOptions: DestinationAttribute[];
 
+  isMappingLoading: boolean = true;
```
Comment (ashwin1111) [inline src/app/shared/components/helper/mapping/generic-mapping-v2/generic-mapping-v2.component.ts:32]: remove

File: src/app/shared/components/helper/mapping/generic-mapping-v2/generic-mapping-v2.component.ts [hunk 4]
```diff
@@ -77,12 +79,12 @@ export class GenericMappingV2Component implements OnInit {
       this.filteredMappings = mappingResponse.results.concat();
       this.filteredMappingCount = this.filteredMappings.length;
       this.totalCount = mappingResponse.count;
-      this.isLoading = false;
+      this.isMappingLoading = false;
```
Comment (ashwin1111) [inline src/app/shared/components/helper/mapping/generic-mapping-v2/generic-mapping-v2.component.ts:82]: make is isloading everywhere and test once

**Pull request merged**